blocks/file-blocks/markdown-edit/extensions.ts

```diff
@@ -124,6 +126,54 @@ export function makeExtensions({
       }
     }),
 
+    EditorView.domEventHandlers({
+      paste(e, view) {
+        const value = e.clipboardData?.items[0];
+        const MAX_IMAGE_SIZE = 5000000;
+        // handle images pasted from the web
+        if (value && value.type === "text/html") {
+          value.getAsString((str) => {
+            const htmlImgRegex = /<img[^>]*src="(?<src>[^"]*)"[^>]*>/gim;
+            const matches = [...str.matchAll(htmlImgRegex)];
+            const images = matches.map((match) => match.groups?.src);
+            if (images) {
+              view.dispatch({
+                changes: {
+                  from: view.state.selection.main.from,
+                  to: view.state.selection.main.to,
+                  insert: images
+                    .filter((image) => image && image.length < MAX_IMAGE_SIZE)
```

Comment:
I guess since this might be a `data:` URL it is consistent with the other case, but I was surprised at first since in other cases the URL length doesn't correspond to the image size. Maybe rename `MAX_IMAGE_SIZE` to `MAX_URL_SIZE` for clarity? Pickiest of nits.

Reply:
yep, that's a good call